pkg/controllers/nodepool/hash/controller.go

```diff
@@ -69,3 +79,37 @@ func (c *Controller) Builder(_ context.Context, m manager.Manager) operatorcontr
 		WithOptions(controller.Options{MaxConcurrentReconciles: 10}),
 	)
 }
+
+func (c *Controller) updateNodeClaimHash(ctx context.Context, np *v1beta1.NodePool) error {
+	ncList := &v1beta1.NodeClaimList{}
+	if err := c.kubeClient.List(ctx, ncList, client.MatchingLabels(map[string]string{v1beta1.NodePoolLabelKey: np.Name})); err != nil {
+		return err
+	}
+
+	errs := make([]error, len(ncList.Items))
+	for i := range ncList.Items {
```

Comment:
Can we either assign to the a variable within the range loop like `for i, nc :=` or (if that's not possible because that would cause addressing issues) can we just assign to a value at the top of the loop like `nc := ncList.Items[i]`? That's going to reduce the verbosity for a lot of the checks that are happening within this range loop